### PR TITLE
AArch64 macOS: Implement omrosdump.c

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements omrosdump.c for AArch64 macOS.

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
I see the warning messages like the following by opening the generated core file with `lldb` against some of the segments in it:

```
% lldb build/macosx-aarch64-normal-server-release/jdk/bin/java -c make/core.20211216.165832.41025.0001.dmp
(lldb) target create "build/macosx-aarch64-normal-server-release/jdk/bin/java" --core "make/core.20211216.165832.41025.0001.dmp"
warning: (arm64e) /Users/openj9/openj9-openjdk-jdk11/make/core.20211216.165832.41025.0001.dmp load command 247 LC_SEGMENT_64 has a fileoff + filesize (0x12d67b238) that extends beyond the end of the file (0xed67b238), the segment will be truncated to match
warning: (arm64e) /Users/openj9/openj9-openjdk-jdk11/make/core.20211216.165832.41025.0001.dmp load command 248 LC_SEGMENT_64 has a fileoff (0x12d67b238) that extends beyond the end of the file (0xed67b238), ignoring this section
Core file '/Users/openj9/openj9-openjdk-jdk11/make/core.20211216.165832.41025.0001.dmp' (arm64) was loaded.
```

I don't know what to do with these warnings, but I see similar messages with OpenJ9 core files on x86-64 macOS.

File: port/osx/omrosdump.c [hunk 1]
```diff
@@ -155,6 +161,30 @@ coredump_to_file(mach_port_t task_port, pid_t pid)
 			goto done;
 		}
 		file_off += sizeof(x86_exception_state_t);
+#elif defined(OMR_ARCH_AARCH64) /* defined(OMR_ARCH_X86) */
+		written = pwrite(corefile_fd, &threads[i].thread_state, sizeof(arm_thread_state64_t), file_off);
```
Comment (keithc-ca) [inline port/osx/omrosdump.c:165]: This whole section need not be AARCH64-specific if, for example, we replace
```
    pwrite(corefile_fd, &threads[i].thread_state, sizeof(arm_thread_state64_t), file_off)
```
with
```
    pwrite(corefile_fd, &threads[i].thread_state, sizeof(threads[i].thread_state), file_off)
```
To achieve this, `neon_state` should be renamed `float_state`.

File: port/osx/omrosdump.c [hunk 2]
```diff
@@ -269,6 +299,34 @@ list_thread_commands(mach_port_t task_port, struct thread_command_full_64 **thre
 			goto done;
 		}
 		threads[i].cmdsize += state_int_count * sizeof(natural_t);
+#elif defined(OMR_ARCH_AARCH64) /* defined(OMR_ARCH_X86) */
```
Comment (keithc-ca) [inline port/osx/omrosdump.c:302]: I think this block can be unified as well, if common names are given to constants, etc.
For example, `THREAD_STATE_COUNT` would be `x86_THREAD_STATE_COUNT` or `ARM_THREAD_STATE64_COUNT` as appropriate.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: port/osx/omrosdump.c (hunk 1 above)
Comment (knn-k) [reply port/osx/omrosdump.c]: Updated as suggested.

File: port/osx/omrosdump.c (hunk 2 above)
Comment (knn-k) [reply port/osx/omrosdump.c]: Updated as suggested.

**Comment by knn-k** [comment]
I ran OpenJ9 sanity.functional to make sure that this PR does not break x86-64_mac, and it was successful.
Internal job/Test_openjdk11_j9_sanity.functional_x86-64_mac_Personal/334/

**Comment by keithc-ca** [comment]
Thanks, @knn-k. The only other (tiny) suggestion I have would be to order architecture considerations alphabetically (so `OMR_ARCH_AARCH64` would be before `OMR_ARCH_X86`), making it easier for people reading the code in the future.

**Comment by knn-k** [comment]
Changed the order of `OMR_ARCH_AARCH64` and `OMR_ARCH_X86` as suggested.

**Review by keithc-ca — APPROVED** [review]

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**